### PR TITLE
HPCC-18819 Add drop zone(s) discovery/listing features to fileservices for ECL programmers.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Implement GetDefaultDropZone() and GetDropZones() functions into Std.File

Update spray_test.ecl to use GetDefaultDropZone() function to eliminate hardwired path usage.

Add dropzone_query_test.ecl to test both new STTD functions

This is a retargeted version of https://github.com/hpcc-systems/HPCC-Platform/pull/12933

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested manually.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please do me a quick check.

File: ecllibrary/std/File.ecl [hunk 1]
```diff
@@ -959,4 +963,30 @@ EXPORT PromoteSuperFileList(set of varstring superNames, varstring addHead='', b
...
+ *                      If more than one Drop Zone
+ *                      process is defined then the first found will
+ *                      be returned; will return an empty string if a Drop Zone
+ *                      cannot be found
+ */
+EXPORT varstring GetDefaultDropZone() :=
+    lib_fileservices.FileServices.GetDefaultDropZone();
+
+ /**
+ * Returns the a dataset with full paths to all Drop Zones
```
Comment (jakesmith) [inline ecllibrary/std/File.ecl]: https://github.com/hpcc-systems/HPCC-Platform/pull/12933#discussion_r325674287

Looks like that comment wasn't addressed.

File: ecllibrary/std/File.ecl [hunk 2]
```diff
@@ -959,4 +963,30 @@ EXPORT PromoteSuperFileList(set of varstring superNames, varstring addHead='', b
 EXPORT varstring GetEspURL(const varstring username = '', const varstring userPW = '') :=
     lib_fileservices.FileServices.GetEspURL(username, userPW);
 
+
+ /**
+ * Returns the path to the default Drop Zone
+ *
+ *
+ * @return              A string containing the path to the default DropZone.
```
Comment (jakesmith) [inline ecllibrary/std/File.ecl]: trivial: everywhere else it comments it says "Drop Zone"..

**Comment by jakesmith** [comment]
@AttilaVamos - 2 trivial issues (one from previous PR)

File: ecllibrary/std/File.ecl (hunk 2 above)
Comment (AttilaVamos) [reply ecllibrary/std/File.ecl]: Changed

**Comment by AttilaVamos** [comment]
@jakesmith both of them are changed.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge

**Pull request merged**